language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/FromJsonGenerator.scala

```diff
@@ -122,7 +123,8 @@ private[inner] object FromJsonGenerator extends StrictLogging {
       .addException(classOf[JsonLfDecoder.Error])
       .addStatement(
         "return jsonDecoder($L).decode(new $T(json))",
-        CodeBlock.join(typeParams.map(t => CodeBlock.of(decodeTypeParamName(t))).asJava, ", "),
+        CodeBlock
+          .join(typeParams.map(t => CodeBlock.of("$L", decodeTypeParamName(t))).asJava, ", "),
```

Comment:
Pull out `"$L"` as the first arg to `CodeBlock.of` is treated as a format string, and will get tripped up if it encounters any `$` chars.